modules/extensions/src/commands/index.ts

```diff
@@ -1,35 +1,151 @@
 import { proxy } from "../util/comlink";
 
-export type Data = Record<string, any>;
-export type Commands = () => Promise<Array<ReturnType<typeof Command>>>;
-export type Run = () => Promise<void>;
-
-export type CommandArgs = {
-  data: Data;
-  commands?: Commands;
-  run?: Run;
+export type CommandFnArgs = {
+  /**
+   * Whether the command is currently active. That is, the user has selected this command in the command bar.
+   *
+   * Subcommands are computed even if the command is not active, so that the command bar can show people helpful suggestions of what they can do next
+   */
+  active: boolean;
+
+  /**
+   * The current search query. This is the text that the user has typed into the command bar.
+   */
+  search: string;
+
+  /**
+   * The current path. This is the path of commands that the user has selected in the command bar.
+   *
+   * The first element of the array is the "root" which contains contextual information about the command. It varies depending on the contribution point.
+   */
+  path: SerializableValue[];
```

Comment:
We talked about this being a class with helper methods, or a tuple of `[SurfaceInfo, ...Array<BaseCommandArgs>]` or the like.